src/player/player.gd

```diff
@@ -56,6 +56,10 @@ func _move(_input_dir:Vector2) -> void:
 	# clear old raycast collison then get new one
 	raycast.force_raycast_update()
 	var next_tile: Object = raycast.get_collider()
+	
+	if next_tile != null and next_tile.is_in_group('npc'):
```

Comment:
We should have npc interactions happen when the interact button is pressed instead of automatically.